contracts/LinkedList.sol

```diff
@@ -56,7 +57,9 @@ contract LinkedList {
         }
 
         uint256 prior = locate(index - 1);
-        list[prior].next = list[list[prior].next].next;
+        uint256 nodeToDelete = list[prior].next;
+        list[prior].next = list[nodeToDelete].next;
+
```

Comment:
you still need to `delete` the item in the map see. https://ethereum.stackexchange.com/questions/10478/clearing-a-custom-struct